src/main/fc/fc_core.c

```diff
@@ -389,13 +402,61 @@ static bool canUpdateVTX(void)
 }
 #endif
 
+// determine if the R/P/Y stick deflection exceeds the specified limit - integer math is good enough here.
+uint8_t areSticksActive(uint8_t stickPercentLimit) {
+    uint8_t deadband;
+    uint8_t stickPercent;
+    bool result = false;
+
+    for (int axis = FD_ROLL; axis <= FD_YAW; axis ++) {
+        if (axis == FD_YAW) {
+            deadband = rcControlsConfig()->yaw_deadband;
+        } else {
+            deadband = rcControlsConfig()->deadband;
+        }
+        if ((rcData[axis] >= PWM_RANGE_MAX) || (rcData[axis] <= PWM_RANGE_MIN)) {
```

Comment:
declare `stickPercent` here.

Reply:
@martinbudden Not sure I understand.  `stickPercent` is declared at the top of the function.  Do you mean declare it inline when assigned?  When there are multiple occurrences I generally declare at the top to avoid having them scattered through the code.

```
bool areSticksActive(uint8_t stickPercentLimit)
{
    uint8_t stickPercent;
    bool ret = false;

    for (int axis = FD_ROLL; axis <= FD_YAW; axis ++) {
        const uint8_t deadband = axis == FD_YAW ? rcControlsConfig()->yaw_deadband : rcControlsConfig()->deadband;
        if ((rcData[axis] >= PWM_RANGE_MAX) || (rcData[axis] <= PWM_RANGE_MIN)) {
            stickPercent = 100;
        } else {
            if (rcData[axis] > (rxConfig()->midrc + deadband)) {
                stickPercent = ((rcData[axis] - rxConfig()->midrc - deadband) * 100) / (PWM_RANGE_MAX - rxConfig()->midrc - deadband);
            } else if (rcData[axis] < (rxConfig()->midrc - deadband)) {
                stickPercent = ((rxConfig()->midrc - deadband - rcData[axis]) * 100) / (rxConfig()->midrc - deadband - PWM_RANGE_MIN);
            } else {
                stickPercent = 0;
            }
        }
        if (stickPercent >= stickPercentLimit) {
            ret = true;
        }
    }
    return ret;
}
```

Reply:
Coding convention for Betaflight is to declare variables when they are first used, except for local scope static variables, which are declared at the beginning of the function. This makes code easier to maintain, and also somewhat easier to read.

Reply:
Ok.  I've always tended to do all my declarations at the top of the function but I guess that's kind of "old school" (like me!).  I don't have a problem changing.  In this case it's a little awkward with the scope of `stickPercent` is in 4 places so I rearranged a little.  It just feels weird to me to simple declarations (without assigning a value) inline.  To me they seem to get lost in the code.  How does this look instead?

```
// determine if the R/P/Y stick deflection exceeds the specified limit - integer math is good enough here.
bool areSticksActive(uint8_t stickPercentLimit)
{
    bool ret = false;

    for (int axis = FD_ROLL; axis <= FD_YAW; axis ++) {
        const uint8_t deadband = axis == FD_YAW ? rcControlsConfig()->yaw_deadband : rcControlsConfig()->deadband;
        uint8_t stickPercent = 0;
        if ((rcData[axis] >= PWM_RANGE_MAX) || (rcData[axis] <= PWM_RANGE_MIN)) {
            stickPercent = 100;
        } else {
            if (rcData[axis] > (rxConfig()->midrc + deadband)) {
                stickPercent = ((rcData[axis] - rxConfig()->midrc - deadband) * 100) / (PWM_RANGE_MAX - rxConfig()->midrc - deadband);
            } else if (rcData[axis] < (rxConfig()->midrc - deadband)) {
                stickPercent = ((rxConfig()->midrc - deadband - rcData[axis]) * 100) / (rxConfig()->midrc - deadband - PWM_RANGE_MIN);
            }
        }
        if (stickPercent >= stickPercentLimit) {
            ret = true;
        }
    }
    return ret;
}
```

Reply:
Looks good to me. The assignment of a default value to `stickPercent` isn't required, but it will be optimised away by the compiler anyway.